src/simcore_service_deployment_agent/git_url_watcher.py

```diff
@@ -354,6 +355,32 @@ async def _clone_and_checkout_repositories(
         )
         await _git_fetch(repo.directory)
 
+    # Checking tags
+    if synced_via_tags:
+        # Sanity check
+        at_least_one_repo_has_tag_regex = any(repo.tags for repo in repos)
+        if not at_least_one_repo_has_tag_regex:
+            raise ConfigurationError(
+                "At least one repo must have a tag-regex specified with tag-sync!"
+            )
+        #
+        each_repo_latest_tags: list[
+            Any
+        ] = await _latest_matching_tag_capture_group_identical_for_repos(repos)
+
+        if not each_repo_latest_tags:
+            log.error("Repos did not match in their latest tag's first capture group!")
```

Comment:
these are maybe a bit too many error/info messages no?
since you raise the exception, and you catch it, this will already show the exception message. no need to add so much messages I think. Also, to the foreign user I am not sure the capture group thing is very helpful. what about:
"the latest tags in the watched repositories are not identical, and synced_via_tag is enabled. TIP: ensure all wactched repositories are tagged identically."

Reply:
adressing this, thx...